pkg/inspector/image-inspector.go

```diff
@@ -541,3 +485,154 @@ func createOutputDir(dirName string, tempName string) (string, error) {
 	}
 	return dirName, nil
 }
+
+// pullExtractAndInspectImage will use containers/image library to pull extract and get the image to be scanned
+// and will inspect it for its metadata.
+// Assumes the image to be pulled is with a transport prefix
+// or if not attempts to add the "docker://" prefix to the image.
+func (i *defaultImageInspector) pullExtractAndInspectImage() (*types.ImageInspectInfo, digest.Digest, error) {
+	policy := &signature.Policy{
+		Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()},
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, "", err
+	}
+	defer policyContext.Destroy()
+
+	srcRef, err := alltransports.ParseImageName(i.opts.Image)
+	if err != nil {
+		// try adding "docker://"
+		i.opts.Image = "docker://" + i.opts.Image
+		srcRef, err = alltransports.ParseImageName(i.opts.Image)
+		if err != nil {
+			return nil, "", fmt.Errorf("Invalid source name %s: %v", i.opts.Image, err)
+		}
+	}
+
+	certPath, err := i.certPath(i.opts.Image)
+	if err != nil {
+		return nil, "", err
+	}
+	sourceCtx := &types.SystemContext{
+		DockerAuthConfig: nil,
+		DockerCertPath:   certPath,
+	}
+
+	if i.opts.DstPath, err = createOutputDir(i.opts.DstPath, "image-inspector-"); err != nil {
+		return nil, "", err
+	}
+	destRef, err := directory.NewReference(i.opts.DstPath)
+	if err != nil {
+		return nil, "", fmt.Errorf("Invalid destination name %s: %v", i.opts.DstPath, err)
+	}
+
+	imagePullAuths, err := i.getAuthConfigs()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	log.Println("Copying image")
```

Comment:
i wish we can controller the verbosity at some point... for now the convention we have for progress messages is to append " ..." at the end ("Copying image ...")